src/config.rs

```diff
@@ -119,25 +118,30 @@ pub struct TlsConfig {
     pub cert: PathBuf,
 }
 
-#[derive(Clone, Derivative)]
-#[derivative(Debug)]
+#[derive(Clone)]
 pub enum Bind {
     Tcp(SocketAddr),
-    Unix(
-        PathBuf,
-        #[derivative(Debug(format_with = "format_permissions"))] u32,
-    ),
+    Unix(PathBuf, u32),
 }
 
-fn format_permissions(permissions: &u32, f: &mut Formatter<'_>) -> std::fmt::Result {
-    write!(f, "0{:o}", permissions)
+impl Debug for Bind {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Bind::Tcp(addr) => f.debug_tuple("Tcp").field(addr).finish(),
+            Bind::Unix(path, permissions) => f
+                .debug_tuple("Unix")
+                .field(path)
+                .field(&format!("0{:0}", permissions))
```

Comment:
Heads up, I think this line is a typo. If you're intending to format the `permissions` field in the usual octal format, the format specifier should be `{:03o}` (at least 3 digits with leading zeros and `o` for octal).